sample/sample-with-di/app/src/main/java/com/dropbox/kaiken/sample/advancedsample/app/AdvancedKaikenSampleApplication.kt

```diff
@@ -1,33 +1,87 @@
 package com.dropbox.kaiken.sample.advancedsample.app
 
 import android.app.Application
-import com.dropbox.kaiken.scoping.AppServices
-import com.dropbox.kaiken.scoping.ScopedServicesProvider
-import com.dropbox.kaiken.scoping.UserServices
+import com.dropbox.kaiken.skeleton.skeleton.core.SkeletonApplication
+import com.dropbox.kaiken.skeleton.skeleton.dagger.SdkSpec
+import com.dropbox.kaiken.skeleton.skeleton.usermanagement.UserManager
+import com.dropbox.kaiken.scoping.AppScope
+import com.dropbox.kaiken.scoping.SingleIn
+import com.dropbox.kaiken.scoping.SkeletonScope
+import com.dropbox.kaiken.scoping.UserScope
+import com.dropbox.kaiken.scoping.cast
+import com.squareup.anvil.annotations.ContributesTo
+import com.squareup.anvil.annotations.MergeComponent
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+import javax.inject.Inject
 
-class AdvancedKaikenSampleApplication : Application(), ScopedServicesProvider {
 
-    private lateinit var appServices: AdvancedKaikenSampleAppServices
+class AdvancedKaikenSampleApplication : SkeletonApplication() {
 
-    // Let's use a map to mimic our "UserManager".
-    private val userServicesMap = mutableMapOf<String, AdvancedKaikenSampleUserServices>()
+    override fun getSdkSpec(): SdkSpec = DaggerSkeletonComponent.factory().create(this) as DaggerSkeletonComponent
 
-    override fun onCreate() {
-        super.onCreate()
+    @Inject
+    lateinit var userFlow: @JvmSuppressWildcards MutableSharedFlow<String>
 
-        appServices = DaggerAdvancedKaikenSampleAppServices.factory().create()
+    @Inject
+    lateinit var userManager: UserManager
 
-        // Let's "log-in" in a user for sample purposes. In a real application this code would definitely not live
+    override fun onCreate() {
+        super.onCreate()
+        appServices.cast<ApplicationInjector>().inject(this)
+        // Let's "log-in" users for sample purposes. In a real application this code would definitely not live
         // here and would be way way more complex
-        val userProfile = UserProfile("awesome_user", "Awesome User")
+        GlobalScope.launch {
```

Comment:
Any reason this needs to be in 3 `GlobalScopes` vs:
```
GlobalScope.launch {
            userFlow.collect {
                userManager.setActiveUser(it)
            }
           userFlow.emit("1")
           userManager.setActiveUser("1")
}

```

Reply:
Yea so if you do it in the same one, collect suspends indefinitely and you never get to the second suspending call. It’s an anti pattern to have multiple in one unless you truly want it to wait for the suspense to finish before going to next one